ydb/docs/en/core/reference/ydb-sdk/coordination.md

```diff
@@ -0,0 +1,314 @@
+        5,              // value to increase semaphore by
+    )
+    ```
+
+    Similar to the session, the `Lease` object also has a context that terminates at one of these moments.
+
+    To cancel waiting for semaphore acquisition, it's sufficient to cancel the passed context `ctx`.
+
+- C++
+
+    ```cpp
+    session
+        .AcquireSemaphore(
+            "my-semaphore",                       // semaphore name
+            TAcquireSemaphoreSettings().Count(5)  // value to increase semaphore by
+        )
+        .ExtractValueSync()
+        .ExtractResult();
+    ```
+
+    When acquiring, you can optionally pass a `TAcquireSemaphoreSettings` structure with the following settings:
+
+    - `Count` - value by which the semaphore is increased upon acquisition.
+    - `Data` - additional data that can be put into the semaphore.
+    - `OnAccepted` - called when the operation is queued. For example, if the semaphore couldn't be acquired immediately.
+
+      - Won't be called if the semaphore is acquired immediately.
+      - It's important to consider that the call can happen in parallel with the `TFuture` result.
+
+    - `Timeout` - maximum time during which the operation can stay in the queue on the server.
+
+      - Operation will return `false` if the semaphore couldn't be acquired within `Timeout` after being added to the queue.
+      - With `Timeout` set to 0, the operation works like `TryAcquire`, i.e., the semaphore will either be acquired atomically and the operation will return `true`, or the operation will return `false` without using queues.
+
+    - `Ephemeral` - if `true`, then the name is an ephemeral semaphore. Such semaphores are automatically created at first `Acquire` and automatically deleted with the last `Release`.
+    - `Shared()` - alias for setting `Count = 1`, acquiring semaphore in shared mode.
+    - `Exclusive()` - alias for setting `Count = max`, acquiring semaphore in exclusive mode. (For semaphores created with limit `Max<ui64>()`.)
+
+{% endlist %}
+
+The taken value of an acquired semaphore can be decreased (but not increased) by calling the `AcquireSemaphore` method again with a smaller value.
+
+### Updating semaphore data {#update-semaphore}
+
+Using the `UpdateSemaphore` method, you can update (replace) the semaphore data that was attached during its creation.
+
+{% list tabs %}
+
+- C++
+
+    ```cpp
+    session
+        .UpdateSemaphore(
+            "my-semaphore",  // semaphore name
+            "updated-data"   // new semaphore data
+        )
+        .ExtractValueSync()
+        .ExtractResult();
+    ```
+
+{% endlist %}
+
+This call doesn't require acquiring the semaphore and doesn't lead to it. If you need only one specific client to update the data, this must be explicitly ensured, for example, by acquiring the semaphore, updating the data, and releasing the semaphore back.
+
+### Getting semaphore data {#describe-semaphore}
+
+{% list tabs %}
+
+- C++
```

Comment:
```suggestion
- Go

    ```go
    description, err := session.DescribeSemaphore(
        "my-semaphore"                                // semaphore name
        options.WithDescribeOwners(true), // to get list of owners
        options.WithDescribeWaiters(true), // to get list of waiters
    )
    ```

- C++
```